torch/distributed/fsdp/_optim_utils.py

```diff
@@ -155,8 +155,7 @@ def _unflatten_optim_state(
             shard_state,
         )
         for optim_state in unflat_param_state:
-            for key in list(optim_state.keys()):
-                state = optim_state[key]
+            for key, state in optim_state.items():
```

Comment:
We cannot do this. Python will complain optim_state is being modified while looping.

Reply:
Oh I see. I'll leave a comment instead